src/bits64/vmx.rs

```diff
@@ -0,0 +1,94 @@
+//! Virtualize processor hardware for multiple software environments using Virtual Machine Extensions.
+
+use crate::bits64::rflags::{self, RFlags};
+use crate::vmx::{Result, VmFail};
+
+/// Helper used to extract VMX-specific Result in accordance with
+/// conventions described in Intel SDM, Volume 3C, Section 30.2.
+#[inline(always)]
+fn vmx_capture_status() -> Result<()> {
+    let flags = unsafe { rflags::read() };
+
+    if flags.contains(RFlags::FLAGS_ZF) {
+        Err(VmFail::VmFailValid)
+    } else if flags.contains(RFlags::FLAGS_CF) {
+        Err(VmFail::VmFailInvalid)
+    } else {
+        Ok(())
+    }
+}
+
+/// Enable VMX operation.
+///
+/// `addr` specifies a 4KB-aligned physical address of VMXON region initialized
+/// in accordance with Intel SDM, Volume 3C, Section 24.11.5.
+#[inline(always)]
```

Comment:
>Is there a reason these functions have to be inline(always)?

You're right, there are no strong reasons to use inline for VMX intrinsics, compared to 3c33d6c29bb546d8774d03825c10a28038c7f7e4.

There are some complexities involved when calling `VMLAUNCH` (it works similarly to `fork(2)` system call and requires you to store some state in order for VMM (host) and VM (guest) to take different branches after it completes), though these can be solved with careful programming, instead of hoping that inlined `vmlaunch()` won't modify any/some registers used by "trampoline" logic.

I've fixed that in db1327be829e2a18b6d876e3953622befbe91d4c.